### PR TITLE
[AutoTVM][TOPI] Fix bifrost spatial packing conv2d auto tune

### PR DESCRIPTION
**Pull request by cchung100m**
Hi @kevinthesun 

Following the issue #5215,  I would appreciate that if you can help to review this PR, thank you.

### CODE REVIEW

File: topi/python/topi/bifrost/conv2d.py [hunk 1]
```diff
@@ -142,13 +142,14 @@ def _schedule_spatial_pack(cfg, s, output, conv, data_vec, kernel_vec):
         s[data_vec].unroll(vw)
 
     if isinstance(kernel_vec.op, tvm.te.ComputeOp) and kernel_vec.name == 'kernel_vec':
+        co, ci, kh, kw, vc = s[kernel_vec].op.axis
```
Comment (kevinthesun) [inline topi/python/topi/bifrost/conv2d.py:145]: Looks like this part share the logic with arm_cpu and mali. We can move this into [conv2d_nchw_spatial_pack compute](https://github.com/apache/incubator-tvm/blob/master/topi/python/topi/arm_cpu/conv2d_spatial_pack.py#L27). An example is in x86 conv2d: https://github.com/apache/incubator-tvm/blob/master/topi/python/topi/x86/conv2d.py#L188-L201

With this method, we can fix mali/arm_cpu/bifrost in one shot.

File: topi/python/topi/bifrost/conv2d.py [hunk 2]
```diff
@@ -370,12 +371,7 @@ def _schedule_winograd(cfg, s, op):
         s[G].compute_inline()
         eps, _, _, _ = s[U].op.axis
         y, _, _, _ = s[padded_kernel].op.axis
-        if autotvm.GLOBAL_SCOPE.in_tuning:
-            # Kernel transformation will be pre-computed during compilation, so we skip
-            # this part to make tuning records correct
-            s[U].pragma(eps, 'debug_skip_region')
-            s[padded_kernel].pragma(y, 'debug_skip_region')
-        else:
+        if not autotvm.GLOBAL_SCOPE.in_tuning:
```
Comment (kevinthesun) [inline topi/python/topi/bifrost/conv2d.py:374]: Also winograd kernel transformation is different from spatial_pack. We need to have a different path to generate placeholder.

File: topi/python/topi/bifrost/conv2d.py (hunk 1 above)
Comment (kevinthesun) [reply topi/python/topi/bifrost/conv2d.py:145]: Actually I'm not quite sure whether we can directly do this kind of placeholder replacement inside schedule. https://github.com/apache/incubator-tvm/pull/5511 for mali might also has issue. Put placeholder replacement in compute would be more convenient.

Comment (cchung100m) [reply topi/python/topi/bifrost/conv2d.py]: Hi @kevinthesun 
Thanks for the review, I move the placeholder replacement to compute.

File: topi/python/topi/arm_cpu/conv2d_spatial_pack.py [hunk 3]
```diff
@@ -267,9 +270,13 @@ def conv2d_spatial_pack_nhwc(cfg, data, kernel, strides, padding, dilation, out_
         data_vec = te.compute(dvshape, lambda n, oho, owo, ohi, owi, ic:
                               data_pad[n][oho*OHI*HSTR+ohi][owo*OWI*WSTR+owi][ic],
                               name='data_vec')
-    kernel_vec = te.compute(kvshape, lambda oco, kh, kw, ic, oci: \
-                            kernel[kh][kw][ic][oco*OCI+oci],
-                            name='kernel_vec')
+
+    if autotvm.GLOBAL_SCOPE.in_tuning:
```
Comment (kevinthesun) [inline topi/python/topi/arm_cpu/conv2d_spatial_pack.py:274]: Change schedule for arm_cpu conv2d as well?

**Comment by kevinthesun** [comment]
Also can you fix winograd kernel replacement?

File: topi/python/topi/bifrost/conv2d.py (hunk 2 above)
Comment (cchung100m) [reply topi/python/topi/bifrost/conv2d.py:374]: Yes. I put the placeholder in `_decl_winograd` to biforst/con2d.py and mali/con2d.py

**Comment by cchung100m** [comment]
Hi @kevinthesun 

Thanks for the prompt reply and reviews. :)

**Comment by kevinthesun** [comment]
Thanks @cchung100m

**Pull request merged**